engine/internal/runtime/manager.go

```diff
@@ -187,15 +187,20 @@ func (m *Manager) PullModel(ctx context.Context, modelID string) error {
 			cleanup()
 			return err
 		}
-		m.autoscaler.Register(modelID, types.NamespacedName{Name: sts.Name, Namespace: sts.Namespace})
-		if sts.Status.ReadyReplicas > 0 {
-			// If this is called before the first cache sync of the reconciler
-			// is complete, the existing statefulset(STS) for the runtime is not
-			// registered in the manager's managed runtime map. If the runtime's
-			// STS is already ready, mark the runtime as ready without waiting
-			// for the reconciler (leader-election component) to process it.
-			m.markRuntimeReady(modelID, client.GetAddress(sts.Name))
+
+		// If this is called before the first cache sync of the reconciler
+		// is complete, the existing statefulset(STS) for the runtime is not
+		// registered in the manager's managed runtime map.
+		//
+		// Run reconcile() here so that if the runtime's STS is already ready,
+		// mark the runtime as ready without waiting for the reconciler
+		// (leader-election component) to process it.
+		if err := m.reconcile(ctx, sts); err != nil {
```

Comment:
If we use reconcile here, this function could update the resource before the leader election if the resource has a deletion timestamp. Since the reconciler and processor's startup timing is different to prevent resource updates, it would be better to process only the runtime registration part without sharing the reconciliation.

Reply:
I see. So wanted to avoid taking mutating actions from a non-leader. Let me withdraw and think.